tests/trial_tests/test_trial.py

```diff
@@ -73,7 +73,7 @@ def test_check_distribution_suggest_uniform(storage_mode: str) -> None:
         study = create_study(storage=storage, sampler=sampler)
         trial = Trial(study, study._storage.create_new_trial(study._study_id))
 
-        with pytest.warns(None) as record:
+        with warnings.catch_warnings(record=True) as record:
```

Comment:
[Question] This might not be related to `strict_equality` since it causes no errors when I revert the change. Could you tell me why you introduced this change?

Reply:
I reverted the change in `test/trial_tests/test_trial.py L.76` and re-run mypy. I got the following results. 

```sh
optuna$ mypy --version
mypy 0.961 (compiled: yes)
optuna$ mypy tests
tests/trial_tests/test_trial.py:76: error: No overload variant of "warns" matches argument type "None"
tests/trial_tests/test_trial.py:76: note: Possible overload variants:
tests/trial_tests/test_trial.py:76: note:     def warns(expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]] = ..., *, match: Union[str, Pattern[str], None] = ...) -> WarningsChecker
tests/trial_tests/test_trial.py:76: note:     def [T] warns(expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]], func: Callable[..., T], *args: Any, **kwargs: Any) -> T
```

I fixed this referring to [this](https://github.com/scikit-learn/scikit-learn/issues/22572).

Reply:
Thank you for your confirmation. I reproduced the error. It was due to  `pytest` and the `checks` CI job cannot detect it because `pytest` is not installed in the job. https://github.com/optuna/optuna/pull/3580 will resolve it, so could you revert the changes in `test_trial.py`, please?